sklearn/decomposition/tests/test_pca.py

```diff
@@ -128,16 +130,20 @@ def test_pca_explained_variance_equivalence_solver(svd_solver):
 
 
 @pytest.mark.parametrize(
-    "X",
+    "make_X",
     [
-        np.random.RandomState(0).randn(100, 80),
-        datasets.make_classification(100, 80, n_informative=78, random_state=0)[0],
+        lambda seed: np.random.RandomState(seed).randn(100, 60),
```

Comment:
The last assertion in this test makes sure that `explained_variance_` is the same as the `n_components` largest eigenvalues of the covariance matrix of `X`.
For two values of `global_random_seed` (17 and 37) this failed with dataset `random-data` and solver `randomized`. Since the randomized SVD is an approximation, a little bit of seed-sensitivity is not suprising if we compare it to the *exact* spectrum of the covariance matrix.
Instead of adjusting the tolerance of the assertion, I slightly reduced the number of features for the `random-data` setting.